### DISCUSSION TITLE
part-ii/2019-104/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#307] [Question comments]
Part II 2019 Number Theory: Paper 3, Section II, I 

https://questions.tripos.org/part-ii/2019-104/

**Comment by callumhobbis** [comment]
<details>
<summary>Sketch solution</summary>

**(a):** Bookwork.

**(b):** Note, working mod $p$, letting $g$ be a primitive root mod $p$, $g^0,\ldots,g^{p-2}$ is a permutation of $1,\ldots,p-1$, so for $k>0$, $S_k\equiv\sum_{x=0}^{p-1}x^k\equiv\sum_{i=0}^{p-2}g^{ik}\equiv(g^k-1)^{-1}((g^k)^{(p-1)}-1)\equiv0\enspace(\text{mod }p)$. For $k=0$ we get $S_0=p$.

**(c):** If $f\equiv0$, then $g\equiv 1$. If $f\not\equiv 0$, then by Fermat's little theorem $g\equiv0$. For the next part, write it as nested sums, and note $\sum_{z\in\{0,1,\ldots,p-1\}}f(x,y,z)=p(ax^2+by^2)+c S_2(z)$, which vanishes mod $p$. For the last part, note if there weren't, then $f\not\equiv 0$ for all $x,y,z\not\equiv 0$ (so $g\equiv0$), and $f(0,0,0)=0$ (so $g\equiv 1$), so that big sum of $g$ would give 1 mod $p$, a contradiction.

</details>
